javascriptv3/example_code/s3/actions/put-bucket-policy.js

```diff
@@ -1,44 +1,78 @@
 // Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 // SPDX-License-Identifier: Apache-2.0
 
-import { fileURLToPath } from "url";
-
 // snippet-start:[s3.JavaScript.policy.putBucketPolicyV3]
-import { PutBucketPolicyCommand, S3Client } from "@aws-sdk/client-s3";
-
-const client = new S3Client({});
+import {
+  PutBucketPolicyCommand,
+  S3Client,
+  S3ServiceException,
+} from "@aws-sdk/client-s3";
 
-export const main = async () => {
+/**
+ * Grant an IAM role GetObject access to all of the objects
+ * in the provided bucket.
+ * @param {{ bucketName: string, iamRoleArn: string }}
+ */
+export const main = async ({ bucketName, iamRoleArn }) => {
+  const client = new S3Client({});
   const command = new PutBucketPolicyCommand({
+    // This is a resource-based policy. For more information on resource-based policies,
+    // see https://docs.aws.amazon.com/IAM/latest/UserGuide/access_policies.html#policies_resource-based.
     Policy: JSON.stringify({
       Version: "2012-10-17",
       Statement: [
         {
-          Sid: "AllowGetObject",
-          // Allow this particular user to call GetObject on any object in this bucket.
           Effect: "Allow",
           Principal: {
-            AWS: "arn:aws:iam::ACCOUNT-ID:user/USERNAME",
+            AWS: iamRoleArn,
           },
           Action: "s3:GetObject",
-          Resource: "arn:aws:s3:::BUCKET-NAME/*",
+          Resource: `arn:aws:s3:::${bucketName}/*`,
         },
       ],
     }),
     // Apply the preceding policy to this bucket.
-    Bucket: "BUCKET-NAME",
+    Bucket: bucketName,
   });
 
   try {
-    const response = await client.send(command);
-    console.log(response);
-  } catch (err) {
-    console.error(err);
+    await client.send(command);
+    console.log(
+      `GetObject access to the bucket "${bucketName}" was granted to the provided IAM role.`,
+    );
+  } catch (caught) {
+    if (
+      caught instanceof S3ServiceException &&
+      caught.name === "NoSuchBucket"
+    ) {
+      console.error(
+        `Error from S3 while setting the bucket policy for the bucket "${bucketName}". The bucket doesn't exist.`,
+      );
+    } else if (caught instanceof S3ServiceException) {
```

Comment:
suggestion: `MalformedPolicy` error handling.

Reply:
That's definitely a better error.